lib/src/util/useToast.ts

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import {useContext} from 'react';
+
+import ToastContext from '../components/NxToast/contexts';
+
+function useToast() {
+  const context = useContext(ToastContext);
+  /**
+   * Base component that renders a standardized toast.
+   * @param message Message to show on the toast.
+   */
+  const showErrorToast = (message: string) => context?.addToast({type: 'error', message});
+  const showSuccessToast = (message: string) => context?.addToast({type: 'success', message});
+  const showWarningToast = (message: string) => context?.addToast({type: 'warning', message});
+  const showInfoToast = (message: string) => context?.addToast({type: 'info', message});
+
+  return { showErrorToast, showSuccessToast, showWarningToast, showInfoToast };
```

Comment:
This API seems a bit non-react-ish.  Was there something that didn't work about exposing a react component that could be rendered wherever and which would internally talk to the context or manage a portal?

Reply:
>This API seems a bit non-react-ish. Was there something that didn't work about exposing a react component that could be rendered wherever and which would internally talk to the context or manage a portal?

Reconfigured here: 
[9540220](https://github.com/sonatype/sonatype-react-shared-components/pull/688/commits/9540220b0a33eafa70152d5de05ed7c8763a6cb2),  [8ce72d2](https://github.com/sonatype/sonatype-react-shared-components/pull/688/commits/8ce72d2768d7f88a888927cd56c8713111658fb0), [99f1577](https://github.com/sonatype/sonatype-react-shared-components/pull/688/commits/99f1577467483d83aae4f84982a8340f3093f45c)